fedimint-api/src/db/mod.rs

```diff
@@ -169,10 +175,86 @@ impl Drop for CommitTracker {
     }
 }
 
+struct IsolatedDatabaseTransaction<'isolated, 'parent: 'isolated> {
+    tx: &'isolated mut DatabaseTransaction<'parent>,
+    prefix: Vec<u8>,
+}
+
+impl<'isolated, 'parent: 'isolated> IsolatedDatabaseTransaction<'isolated, 'parent> {
+    pub fn new(
+        dbtx: &'isolated mut DatabaseTransaction<'parent>,
+        module_instance_id: ModuleInstanceId,
+    ) -> IsolatedDatabaseTransaction<'isolated, 'parent> {
+        let mut prefix_bytes = vec![MODULE_GLOBAL_PREFIX];
+        module_instance_id
+            .consensus_encode(&mut prefix_bytes)
+            .expect("Error encoding module instance id as prefix");
+        IsolatedDatabaseTransaction {
+            tx: dbtx,
+            prefix: prefix_bytes,
+        }
+    }
+}
+
+#[async_trait]
+impl<'isolated, 'parent> IDatabaseTransaction<'isolated>
+    for IsolatedDatabaseTransaction<'isolated, 'parent>
+{
+    async fn raw_insert_bytes(&mut self, key: &[u8], value: Vec<u8>) -> Result<Option<Vec<u8>>> {
+        let mut key_with_prefix = self.prefix.clone();
+        key_with_prefix.append(&mut key.to_vec());
```

Comment:
Calling to `.to_vec()` just to append it to another vec seems like a waste (even if compiler can optimize). `.extend_from_slice`?

Reply:
Changed to use extend_from_slice